### ISSUE TITLE
Excel Online: calculation state can get stuck in Pending state when workbook contains charts

### ISSUE THREAD

**Issue by wh1t3cAt1k** [OfficeDev/office-js#4760]
# Provide required information needed to triage your issue

This is, I suspect, an exact copy (same scenario, same problem, same context) of https://github.com/OfficeDev/office-js/issues/2883, which I know has been closed after actions on the Desktop platforms.

However, while verifying the fix, Velixo discovered that **in Excel Online, the problem still remains**. 
When we remove all chart objects from the workbook, the problem disappears.

Great job on the issue above guys, could we just ask you to verify the same root cause in Excel Online and do a follow-up fix for this platform if possible? Thank you!

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Office on the Web
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: latest
* Operating System: Windows 11
* Browser (if using Office on the web): Edge
 
## Expected behavior

Polling calculation state of the `context.application` should resolve from `Pending` to `Calculating` / `Done` eventually.

## Current behavior

When 
The root cause I think is the same as #2883.

## Steps to reproduce

Please check with the Desktop partner team internally whether the same steps as #2883 can be used to replicate. If not, probably we can give you (more complex) repro steps using the Velixo add-in!

[Velixo internal tracking item](https://3.basecamp.com/3496385/buckets/16689259/card_tables/cards/7684748963)

**Comment by wh1t3cAt1k** [comment]
As promised, I am sharing the replication video. Credentials will be provided in the email thread.

https://www.loom.com/share/f6892c324828401ca53dbed847d03132

Add-in to use: https://appsource.microsoft.com/en-us/product/office/wa200002311?tab=overview
ScriptLab snippet gist: https://gist.github.com/agazukin/cebcf4c5b184f10d608b73f7855505aa
Excel file to use: [calculation-state-online 1.xlsx](https://github.com/user-attachments/files/16543884/calculation-state-online.1.xlsx)

**Notes**:

* It is very important to **first** refresh the workbook using Velixo, and **only then** run the ScriptLab snippet to catch the stuck calculation state. Because of some conflict, the two add-ins **cannot** be run in parallel.
* For the ScriptLab gist, don't forget to click Run button and then enter a random value in any cell to trigger the Calculated event.
* In the first part of the video, after I enter 42 in a random cell, you can see how the chart refreshes and the calculation state gets stuck in Pending forever.
* In the second part of the video, after I removed charts with non-contiguous source ranges, the issue disappeared and the calculation settled in Done. This IMO proves it's the same issue as #2883.

**Comment by XuanZhouMSFT** [comment]
@wh1t3cAt1k Thanks for providing the video.But It seems there's something wrong. While I see the video, before 3:00, it is always hangs as the below image:
<img width="1177" alt="image" src="https://github.com/user-attachments/assets/a777eea8-01e1-440f-97d8-2c31541b416c">
And after, it will always hangs as the below image:
<img width="1172" alt="image" src="https://github.com/user-attachments/assets/ca6a2f65-94cb-44ce-8fb3-5fd5ebdfd648">

Could you help to double check with it?

**Comment by wh1t3cAt1k** [comment]
@XuanZhouMSFT thank you, something went wrong with my Loom on that one. I replaced the video with an older one, the sound is awful but at least the picture is moving and you can still make sense of what's going on. Please let me know if this one works and allows you to repro.

**Comment by wh1t3cAt1k** [comment]
@XuanZhouMSFT hoping for the investigation results on your side!

**Comment by jaiswalshubh1** [comment]
@wh1t3cAt1k - apologies for the long silence after you supplied a video, a workbook, a Script Lab snippet and a re-recording when the first video did not play. You asked us to verify the same root cause on the web, and we did - and then never told you the result.

Where it actually stands: the cause is understood. The calculation state can stick at `Pending` when a chart in the workbook references a volatile function; a minimal reproduction is a few cells of `=rand()`, a chart over them, then polling `calculationState`. The fix that shipped for the related report you referenced, #2883, covered the desktop platforms only and was never extended to Excel on the web.

It was later reassessed and did not clear the bar for a fix, on the basis that it is long-standing rather than a recent regression. You were told none of that, which is the part I am sorry about - you had effectively already done the investigation for us.

If the stuck `Pending` state is still affecting your product, say so here and I will make sure that impact is attached to the record rather than left on this thread.

cc: @kustonaut
